### DISCUSSION TITLE
How do I correctly pass plugin options?

### DISCUSSION THREAD

**Discussion by vypxl** [NvChad/NvChad#1714] [Q&A] [answered]
I want to use [andweeb/presence.nvim](/andweeb/presence.nvim)
and set the config option `main_image = "file"`.

How do I do this? I tried like 10 different things but none worked.
I feel like the correct way should be something like this right?

```lua
-- in chadrc
M.plugins = {
  ...
  ["andweeb/presence.nvim"] = {
    main_image = "file" 
  },
  ...
```

**Comment by siduck** [comment, marked as answer]
@vypxl 

```lua
  ["andweeb/presence.nvim"] = {
       config  = function() 
           require("presence"):setup({
                  main_image = "file" 
           })
       end
  },
  ```

**Reply by vypxl** [reply]
Oh my god
I tried this approach as well
The thing is, I used `.setup()` instead of `:setup()` :facepalm: 
The error message resulting from this gave me headaches

Lua at its best

Thank you!

**Reply by siduck** [reply]
@vypxl setup function gets called before the plugin loads so it wont work :p
